[#154643827] create SearchgovDomain model

MothOnMars opened the pull request.

noremmie commented on db/migrate/20180408135739_create_searchgov_domains.rb:
```diff
@@ -0,0 +1,15 @@
+      t.timestamps null: false
+    end
+
+    add_index :searchgov_domains, :domain, length: 100, unique: true
```
Is there any reason why this shouldn't use `length: 255` since the`domain` column is that big?

MothOnMars replied:
Yes, MySQL barfs when you try to create a unique index on 255 char. I considered a) making the domains column only 100 char, but *technically* a domain name can be up to 255 char. I figured that the chances of our domains even hitting 100 char are slim enough that I'm not worried about false positives on a unique check for two different 100+ char domains with the same first 100 characters.

noremmie replied:
Ah... that makes sense. I had checked the index on `searchgov_urls.url` and saw that it was 255 characters, but that index isn't unique. So it wasn't right for me to use that as a comparison 👍.

noremmie commented on spec/support/fetchable_behavior.rb:
```diff
@@ -115,4 +115,16 @@
+  describe '#indexed?' do
+    subject(:indexed) { record.indexed? }
+
+    context 'when the last_crawl_status = "OK"' do
```
Minor nit: it might be nice to have a test for when `#indexed?` returns false.

MothOnMars replied:
can do

noremmie commented on app/models/searchgov_url.rb:
```diff
@@ -15,15 +15,26 @@ class SearchgovUrl < ActiveRecord::Base
   before_destroy :delete_document
 
+  belongs_to :searchgov_domain
+  counter_culture :searchgov_domain, column_name: 'urls_count'
```
I read up on the `counter_culture` gem since I had never heard of it. I think I get what's going on here. This is what's responsible for keeping `searchgov_domains.urls_count` and `searchgov_domains.unfetched_urls_count` up to date 👍.

MothOnMars commented:
@noremmie , apologies for those extra `.orig` files. I pushed a minor late-game amendment to this commit that I clearly didn't review properly. Fixing...

MothOnMars commented:
I've removed those `.orig` files and added the spec. There will be a follow-up story to add the super admin page. I was doing those in tandem, but I decided I should split them up to simplify the review & testing. I will also need to create the domain records & update the existing searchgov url records with the association, which I plan to do after this is merged and before the super admin pages are set up.

noremmie commented:
Sounds good, and LGTM 👍
